src/PowerShellEditorServices/Services/Extension/Handlers/InvokeExtensionCommandHandler.cs

```diff
@@ -4,40 +4,28 @@
 using System.Threading;
 using System.Threading.Tasks;
 using MediatR;
-using Microsoft.Extensions.Logging;
 using Microsoft.PowerShell.EditorServices.Extensions;
 
 namespace Microsoft.PowerShell.EditorServices.Services.Extension
 {
     internal class InvokeExtensionCommandHandler : IInvokeExtensionCommandHandler
     {
-        private readonly ILogger<InvokeExtensionCommandHandler> _logger;
         private readonly ExtensionService _extensionService;
         private readonly EditorOperationsService _editorOperationsService;
 
         public InvokeExtensionCommandHandler(
-            ILoggerFactory factory,
             ExtensionService extensionService,
-            EditorOperationsService editorOperationsService
-            )
+            EditorOperationsService editorOperationsService)
         {
-            _logger = factory.CreateLogger<InvokeExtensionCommandHandler>();
             _extensionService = extensionService;
             _editorOperationsService = editorOperationsService;
         }
 
         public async Task<Unit> Handle(InvokeExtensionCommandParams request, CancellationToken cancellationToken)
         {
-            // We can now await here because we handle asynchronous message handling.
-            EditorContext editorContext =
-                _editorOperationsService.ConvertClientEditorContext(
-                    request.Context);
-
-            await _extensionService.InvokeCommandAsync(
-                request.Name,
-                editorContext).ConfigureAwait(false);
-
-            return await Unit.Task.ConfigureAwait(false);
```

Comment:
No idea why this was the old return type. Almost everything else using the same API returns `Unit.Value` (which seems the right thing to return).

Reply:
Ah, clarification on that "almost" it's that `async Task<Unit>` with an `await` needs to return `Unit.Value`, and if it's not `async` then it's a `Unit.Task` that needs to be returned.